code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -156,8 +156,60 @@
 		//G.reagents.trans_to(src,G.amount_per_transfer_from_this)
 	else if(istype(O,/obj/item/weapon/grab))
 		var/obj/item/weapon/grab/G = O
-		to_chat(user, "<span class='alert'>This is ridiculous. You can not fit \the [G.affecting] in this [src].</span>")
-		return 1
+		if(!istype(src, /obj/machinery/kitchen_machine/oven) && !istype(src, /obj/machinery/kitchen_machine/grill))
```

Comment:
Put  `burn_face` proc call here, declare a `burn_face` proc on the `kitchen_machine` path, then overload the `burn_face` proc as appropriate on the oven and grill. You can then do a similar thing on the cooker path. You can put the logic for "You can't do this" in the base of the `burn_face` for a "user-friendly" stop trying.

If you don't do that, this won't get merged.